src/components/ModalBuyMeal/ModalBuyMealPool.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import styles from './styles.module.scss';
+import { useModalPaymentDispatch } from '../../utilities/hooks/ModalPaymentContext/context';
+import {
+  SET_MODAL_VIEW,
+  SET_AMOUNT,
+} from '../../utilities/hooks/ModalPaymentContext/constants';
+import { useTranslation } from 'react-i18next';
+import walletImage from './wallet.png';
+import cardImage from './card.png';
+import ReactPixel from 'react-facebook-pixel';
+
+export interface Props {
+  purchaseType: string;
+  sellerId: string;
+  sellerName: string;
+  costPerMeal: number;
+}
+
+export const Modal = (props: Props) => {
+  const { t } = useTranslation();
+  const dispatch = useModalPaymentDispatch();
```

Comment:
Watch out, I think you and @bvillaroman will have to reconcile who merges first, as I think after his change goes in, this will have to be `useModalPaymentDispatch(null)`